File: Plugins/VanillaBasics/src/main/java/org/tobi29/scapes/vanilla/basics/VanillaBasicsMetals.java
package org.tobi29.scapes.vanilla.basics;

class VanillaBasicsMetals {
    static void registerMetals(VanillaBasics plugin) {
        plugin.c.metal(m -> {
            m.id = m.name = m.ingotName = "Tin";
            m.meltingPoint = 231.0;
            m.toolEfficiency = 0.1;
            m.toolStrength = 2.0;
            m.toolDamage = 0.01;
            m.toolLevel = 10;
            m.r = 1.0f;
            m.g = 1.0f;
            m.b = 1.0f;
        });
        plugin.c.metal(m -> {
            m.id = m.name = m.ingotName = "Zinc";
            m.meltingPoint = 419.0;
            m.toolEfficiency = 1.0;
            m.toolStrength = 4.0;
            m.toolDamage = 0.004;
            m.toolLevel = 10;
            m.r = 1.0f;
            m.g = 0.9f;
            m.b = 0.9f;
        });
        plugin.c.metal(m -> {
            m.id = m.name = m.ingotName = "Bismuth";
            m.meltingPoint = 271.0;
            m.toolEfficiency = 1.0;
            m.toolStrength = 4.0;
            m.toolDamage = 0.004;
            m.toolLevel = 10;
            m.r = 0.8f;
            m.g = 0.9f;
            m.b = 0.9f;
        });
        plugin.c.metal(m -> {
            m.id = m.name = m.ingotName = "Copper";
            m.meltingPoint = 1084.0;
            m.toolEfficiency = 6.0;
            m.toolStrength = 8.0;
            m.toolDamage = 0.001;
            m.toolLevel = 20;
            m.r = 0.8f;
            m.g = 0.2f;
            m.b = 0.0f;
        });
        plugin.c.metal(m -> {
            m.id = m.name = m.ingotName = "Iron";
            m.meltingPoint = 1538.0;
            m.toolEfficiency = 30.0;
            m.toolStrength = 12.0;
            m.toolDamage = 0.0001;
            m.toolLevel = 40;
            m.r = 0.7f;
            m.g = 0.7f;
            m.b = 0.7f;
        });
        plugin.c.metal(m -> {
            m.id = m.name = m.ingotName = "Silver";
            m.meltingPoint = 961.0;
            m.toolEfficiency = 1.0;
            m.toolStrength = 4.0;
            m.toolDamage = 0.004;
            m.toolLevel = 10;
            m.r = 0.9f;
            m.g = 0.9f;
            m.b = 1.0f;
        });
        plugin.c.metal(m -> {
            m.id = m.name = m.ingotName = "Gold";
            m.meltingPoint = 1064.0;
            m.toolEfficiency = 0.1;
            m.toolStrength = 2.0;
            m.toolDamage = 0.01;
            m.toolLevel = 10;
            m.r = 0.9f;
            m.g = 0.9f;
            m.b = 1.0f;
        });
        plugin.c.alloy(m -> {
            m.id = m.name = m.ingotName = "Bronze";
            m.toolEfficiency = 10.0;
            m.toolStrength = 10.0;
            m.toolDamage = 0.0005;
            m.toolLevel = 20;
            m.r = 0.6f;
            m.g = 0.4f;
            m.b = 0.0f;
            m.ingredients.put("Tin", 0.25);
            m.ingredients.put("Copper", 0.75);
        });
        plugin.c.alloy(m -> {
            m.name = m.ingotName = "Bismuth Bronze";
            m.id = "BismuthBronze";
            m.toolEfficiency = 10.0;
            m.toolStrength = 10.0;
            m.toolDamage = 0.0005;
            m.toolLevel = 20;
            m.r = 0.6f;
            m.g = 0.4f;
            m.b = 0.0f;
            m.ingredients.put("Bismuth", 0.2);
            m.ingredients.put("Zinc", 0.2);
            m.ingredients.put("Copper", 0.6);
        });
    }
}

File: Plugins/VanillaBasics/src/main/kotlin/org/tobi29/scapes/vanilla/basics/VanillaBasicsMetals.kt
/*
 * Copyright 2012-2017 Tobi29
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.tobi29.scapes.vanilla.basics

internal fun VanillaBasics.registerMetals() {
    metal {
        id = "Tin"
        name = "Tin"
        ingotName = "Tin"
        meltingPoint = 231.0
        toolEfficiency = 0.1
        toolStrength = 2.0
        toolDamage = 0.01
        toolLevel = 10
        r = 1.0f
        g = 1.0f
        b = 1.0f
    }
    metal {
        id = "Zinc"
        name = "Zinc"
        ingotName = "Zinc"
        meltingPoint = 419.0
        toolEfficiency = 1.0
        toolStrength = 4.0
        toolDamage = 0.004
        toolLevel = 10
        r = 1.0f
        g = 0.9f
        b = 0.9f
    }
    metal {
        id = "Bismuth"
        name = "Bismuth"
        ingotName = "Bismuth"
        meltingPoint = 271.0
        toolEfficiency = 1.0
        toolStrength = 4.0
        toolDamage = 0.004
        toolLevel = 10
        r = 0.8f
        g = 0.9f
        b = 0.9f
    }
    metal {
        id = "Copper"
        name = "Copper"
        ingotName = "Copper"
        meltingPoint = 1084.0
        toolEfficiency = 6.0
        toolStrength = 8.0
        toolDamage = 0.001
        toolLevel = 20
        r = 0.8f
        g = 0.2f
        b = 0.0f
    }
    metal {
        id = "Iron"
        name = "Iron"
        ingotName = "Iron"
        meltingPoint = 1538.0
        toolEfficiency = 30.0
        toolStrength = 12.0
        toolDamage = 0.0001
        toolLevel = 40
        r = 0.7f
        g = 0.7f
        b = 0.7f
    }
    metal {
        id = "Silver"
        name = "Silver"
        ingotName = "Silver"
        meltingPoint = 961.0
        toolEfficiency = 1.0
        toolStrength = 4.0
        toolDamage = 0.004
        toolLevel = 10
        r = 0.9f
        g = 0.9f
        b = 1.0f
    }
    metal {
        id = "Gold"
        name = "Gold"
        ingotName = "Gold"
        meltingPoint = 1064.0
        toolEfficiency = 0.1
        toolStrength = 2.0
        toolDamage = 0.01
        toolLevel = 10
        r = 0.9f
        g = 0.9f
        b = 1.0f
    }
    alloy {
        id = "Bronze"
        name = "Bronze"
        ingotName = "Bronze"
        toolEfficiency = 10.0
        toolStrength = 10.0
        toolDamage = 0.0005
        toolLevel = 20
        r = 0.6f
        g = 0.4f
        b = 0.0f
        ingredients.put("Tin", 0.25)
        ingredients.put("Copper", 0.75)
    }
    alloy {
        id = "BismuthBronze"
        name = "Bismuth Bronze"
        ingotName = "Bismuth Bronze"
        toolEfficiency = 10.0
        toolStrength = 10.0
        toolDamage = 0.0005
        toolLevel = 20
        r = 0.6f
        g = 0.4f
        b = 0.0f
        ingredients.put("Bismuth", 0.2)
        ingredients.put("Zinc", 0.2)
        ingredients.put("Copper", 0.6)
    }
}
